node/node.go

```diff
@@ -1291,13 +1292,49 @@ func (fs *firstSessionErrorTracker) SessionError(err error) {
 	fs.mu.Unlock()
 }
 
+// SessionClosed returns an error if we haven't yet established a session, and
+// we get a gprc error as a result of an X509 failure.
 func (fs *firstSessionErrorTracker) SessionClosed() error {
 	fs.mu.Lock()
 	defer fs.mu.Unlock()
-	// unfortunately grpc connection errors are type grpc.rpcError, which are not exposed, and we can't get at the underlying error type
-	if !fs.pastFirstSession && grpc.Code(fs.err) == codes.Internal &&
-		strings.HasPrefix(grpc.ErrorDesc(fs.err), "connection error") && strings.Contains(grpc.ErrorDesc(fs.err), "transport: x509:") {
-		return fs.err
+
+	// if we've successfully established at least 1 session, never return
+	// errors
+	if fs.pastFirstSession {
+		return nil
 	}
-	return nil
+
+	// get the GRPC status from the error, because we only care about GRPC
+	// errors
+	grpcStatus, ok := status.FromError(fs.err)
+	// if this isn't a GRPC error, it's not an error we return from this method
+	if !ok {
+		return nil
+	}
+
+	// additionally, the error we're looking for is an internal error
+	if grpcStatus.Code() != codes.Internal {
```

Comment:
Wondering; does it return a specific `.Code()` that could be useful for detection (in addition to string-matching)?